### ISSUE TITLE
[js/html] syntax highlighting not correct for javascript multiline comments (/* */)   in html file

### ISSUE THREAD

**Issue by alexr00** [textmate/html.tmbundle#99]
_From @allan-simon on January 13, 2019 19:53_

- VSCode Version: 1.30.2
- OS Version:  Windows 10

Steps to Reproduce:

1. open a .html file 
2. write 

```
<button
      onclick="
            /*
                var toto = 'hey i look like normal javascript code';
            */
      "
>
     clickme
</button>
```
the first `/*`  is correctly colored (actually the whole first line is correctly colored if you put additional text) 
but the next lines are not.

Does this issue occur when all extensions are disabled?: Yes

_Copied from original issue: Microsoft/vscode#66457_

**Comment by alexr00** [comment]
_From @vscodebot[bot] on January 13, 2019 19:53_

(Experimental duplicate detection)
Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
- [VSCode Incorrect Syntax Highlight in HTML file (#65154)](https://www.github.com/Microsoft/vscode/issues/65154)

**Comment by alexr00** [comment]
_From @mjbvz on January 14, 2019 16:32_

@aeschli I think this is an html grammar specific issue related to how is treats event handler attributes as JavaScript. I haven't been able to reproduce in a normal js file

It looks like multiline matches do not work for the embedded JS in attributes

**Comment by alexr00** [comment]
_From @mjbvz on January 14, 2019 19:16_

Caused by this line: https://github.com/Microsoft/vscode/blob/32d22dbace8f2deed5dfb44e19f813f76ff1d0f0/extensions/html/syntaxes/html.tmLanguage.json#L225

I believe the problem is that `matches` will only apply to a single line at a time

**Comment by alexr00** [comment]
_From @allan-simon on January 15, 2019 17:8_

thanks for starting looking into this bug :)  I'm teaching to code to a friend of mine and right now we don't have seen yet functions etc.  so we're currently coding inside the "onclick" etc.  so this was bug was quite surprising for her ^^

**Comment by infininight** [comment]
Solved in 0c3d5ee54de3a993f747f54186b73a4d2d3c44a2.

The thorny issue here is to ensure we respect the attribute bounds, thusly matching the lines before passing them to the JavaScript grammar to prevent overflow there. This meant its block comment rule started but couldn't continue. This has been solved by adding a custom block comment rule.

Tested with:

```html
<button
      onclick="
            /*
                var toto = 'hey i look like normal javascript code';
            */
      "
>
     clickme
</button>


<button
      onclick='
            /*
                var toto = "hey i look like normal javascript code";
            */
      '
>
     clickme
</button>

<button
      onclick="
            /*
                var toto = " bar="foo"
>
     clickme
</button>

<button
      onclick='
            /*
                var toto = ' bar='foo'
>
     clickme
</button>
```

**Comment by allan-simon** [comment]
thanks a lot @infininight
